core/src/main/java/com/crawljax/core/CandidateElementExtractor.java

```diff
@@ -340,7 +340,7 @@ private boolean isExternal(String href) {
 	 * @return true if href has the pdf or ps pattern.
 	 */
 	private boolean isFileForDownloading(String href) {
-		final Pattern p = Pattern.compile(".+.pdf|.+.ps|.+.zip|.+.mp3");
+		final Pattern p = Pattern.compile(".+\\.pdf$|.+\\.ps$|.+\\.zip$|.+\\.mp3$");
```

Comment:
Is using the end anchor the right choice? What if there's a param after the file name? Ex: `href=Fred.pdf?foo=bar`

Reply:
Yeah, that might be too much restrictive.

Reply:
I had never thought of that!! Maybe the regex = __`.+\.(?:pdf|ps|zip|mp3)(?:$|\?.+)`__ will be better?

BTW, I want to ask some question about this issue (`href=Fred.pdf?foo=bar`).
Are there any situations that the param will be after the file name? Why?

Reply:
I'd need to do some testing but I think end a chore or question mark is still a mistake. I believe the etc defines other separators vs question mark (though it us by far the most proffered by the industry). Perhaps word boundary? `\w` IIRC
I can check on regexplanet in a few hours.

Reply:
I think the only separator for the path and query component (per rfc) is the question mark, there are some frameworks that use `;` to inject the session ID though, we could include that as well but more chars will lead to wrong exclusions.